packages/flutter_tools/templates/create/pubspec.yaml.tmpl

```diff
@@ -1,6 +1,14 @@
 name: {{projectName}}
 description: {{description}}
 
+# The following defines the version and build number for your application.
+# A version number is three numbers separated by dots, like 1.2.43
+# followed by an optional build number separated by a +.
+# Both, version and build number, can be overwritten with flutter build command.
+# Read more about versioning at semver.org. Example:
+# version: 1.2.43+177
+version: 1.0.0+1
```

Comment:
Maybe rename the flutter build command flag to be `--build-version` to match this?